wolfssl/ssl.h

```diff
@@ -1367,10 +1383,12 @@ WOLFSSL_API WOLFSSL_SESSION* wolfSSL_get1_session(WOLFSSL* ssl);
                            /* what's ref count */
 
 WOLFSSL_API WOLFSSL_X509* wolfSSL_X509_new(void);
-#if defined(OPENSSL_EXTRA_X509_SMALL) || defined(OPENSSL_ALL)
+#if defined(OPENSSL_EXTRA_X509_SMALL) || defined(OPENSSL_EXTRA)
```

Comment:
why change from openssl_all to openssl_extra here? The define between the two is already grey ..but having the compiled functions switch periodically could make things not work for builds that used to use the functions.

Reply:
My understanding of the defines is this: OPENSSL_EXTRA_X509_SMALL < OPENSSL_EXTRA < OPENSSL_ALL. With each level being a superset of the previous. This define was including the functions for OPENSSL_EXTRA_X509_SMALL but not for OPENSSL_EXTRA which is both confusing and doesn't follow the rule of each level including the previous. I changed this to be more consistent on what is and isn't available.

Reply:
My concern is that the OPENSSL_EXTRA_X509_SMALL build changes else where in the code and this change from OPENSSL_ALL to OPENSSL_EXTRA might not be related to HAPROXY, but have the potential for causing issues with other existing builds. Checking the OPENSSL_EXTRA_X509_SMALL build after this PR the code size has increased some, which maybe unavoidable. Yes, the OPENSSL_EXTRA_X509_SMALL build is a subset of OPENSSL_EXTRA but it was also fine tuned initially for a specific use case with not a lot of memory (including memory taken up by code compiled into a static library).

It would be better I think to avoid changing what functions are compiled with OPENSSL_ALL vs OPENSSL_EXTRA vs OPENSSL_EXTRA_X509_SMALL if possible with the HAPROXY port update and have a separate PR that addresses refactoring that code.

Reply:
I don't think I'm changing too much what gets included with `OPENSSL_EXTRA_X509_SMALL`. Many of the changes I made were for consistency. For example the reference counters were being inconsistently compiled in. I don't think these changes will break other builds (it might even address some issues).

> separate PR that addresses refactoring that code.

I prefer addressing small changes immediately as they come up. I think that these inconsistencies can be addressed directly by this PR.